plugin.video.embycon/resources/lib/item_functions.py

```diff
@@ -347,22 +367,50 @@ def add_gui_item(url, item_details, display_options, folder=True):
     addCounts = display_options["addCounts"]
     if addCounts and item_details.unwatched_episodes != 0:
         countsAdded = True
-        listItemName = listItemName + (" (%s)" % item_details.unwatched_episodes)
+        list_item_name = list_item_name + (" (%s)" % item_details.unwatched_episodes)
 
     addResumePercent = display_options["addResumePercent"]
     if (not countsAdded
             and addResumePercent
             and cappedPercentage not in [0, 100]):
-        listItemName = listItemName + (" (%s%%)" % cappedPercentage)
+        list_item_name = list_item_name + (" (%s%%)" % cappedPercentage)
 
     subtitle_available = display_options["addSubtitleAvailable"]
     if subtitle_available and item_details.subtitle_available:
-        listItemName += " (cc)"
+        list_item_name += " (cc)"
+
+    if item_details.item_type == "Program":
+        start_time = datetime_from_string(item_details.program_start_date)
+        end_time = datetime_from_string(item_details.program_end_date)
+
+        duration = (end_time - start_time).total_seconds()
+        time_done = (datetime.now() - start_time).total_seconds()
+        percentage_done = (float(time_done) / float(duration)) * 100.0
+        cappedPercentage = int(percentage_done)
+
+        start_time_string = start_time.strftime("%H:%M")
+        end_time_string = end_time.strftime("%H:%M")
+
+        item_details.duration = int(duration)
+        item_details.resume_time = int(time_done)
+
+        list_item_name = (item_details.program_channel_name +
+                          " - " + list_item_name +
+                          " - " + start_time_string + " to " + end_time_string +
+                          " (" + str(int(percentage_done)) + "%)")
+
+        time_info = "Start : " + start_time_string + "\n"
+        time_info += "End : " + end_time_string + "\n"
+        time_info += "Complete : " + str(int(percentage_done)) + "%\n"
+        if item_details.plot:
+            item_details.plot = time_info + item_details.plot
+        else:
+            item_details.plot = time_info
 
     if kodi_version > 17:
-        list_item = xbmcgui.ListItem(listItemName, offscreen=True)
+        list_item = xbmcgui.ListItem(list_item_name, offscreen=True)
     else:
-        list_item = xbmcgui.ListItem(listItemName, iconImage=thumbPath, thumbnailImage=thumbPath)
+        list_item = xbmcgui.ListItem(list_item_name, iconImage=thumbPath, thumbnailImage=thumbPath)
```

Comment:
Fyi this will probably be dropped completely in v19

Reply:
ListItems are going away?

Reply:
Iconimage and thumbnailImage kwargs

Reply:
that's cool, I use setArt() to do a full artwork setup a few lines down so this can just be removed.

This will be addressed in the next major version for Kodi 19, this will be a major version bump and a big rewrite to fix all the Python 3 issues the current add-on has.